### PR TITLE
feat: add gh workflow for auto release ticket creation

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### Description Of Changes
Addresses https://sendbird.atlassian.net/browse/CLNP-176 
This PR is a very beginning the project of automating our release process. Two small things are done in this PR, 

- [x] Added a new gh workflow file to receive a signal from outside of Github  e.g. Slack / GIthub CLI / Other 3party tools 
  👉🏻 This enables to create a release Jira ticket without navigating to Jira board or Github. 
(Got an idea from [a proposal](https://sendbird.atlassian.net/wiki/spaces/UIKitreact/pages/2101838231/Release+Automation#Concrete-Proposal-(June-23---2023)) from @sravan-s, see the Slack bot section)
Once this workflow file is merged into `main` branch, running this below command from a Slack channel will create a ticket.  Note that the command accepts only `branch_name` at the moment. 
```
/github workflow dispatch sendbird/sendbird-uikit-react --workflow release-ticket-reaction.yml -i branch_name=your-branch-name
```
The newly created ticket can be found in #sdk-release-approver channel in Slack. You can ask the approvers to check the ticket in the channel.
But yeah, the cmd is too long to remember. So I'm thinking of creating a short custom command in Slack for the sake of easy-use.

- [x] Modified the Circle CI config to take an advantage of existing Jira ticket creation pipeline 
 👉🏻 got the most of logic from the chat ios repo; https://github.com/sendbird/chat-ios/blob/develop/.github/workflows/pr-comment-bot.yml#L90

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,10 +61,68 @@ jobs:
           key: uikit-yarn-dependencies-{{ checksum "yarn.lock" }}
           paths:
             - ./.yarn/cache
+  create-ticket:
+    macos:
+      xcode: "14.1"
```
Comment (sravan-s) [inline .circleci/config.yml:66]: this is for code-sign in ios, probably not necessary for us

File: .circleci/config.yml [hunk 2]
```diff
@@ -43,10 +61,68 @@ jobs:
...
+    steps:
+      - checkout
+      - clone_deployment_script
+      - run:
+          name: Extract release version
+          command: |
+            release_version=$(python3 ~/sdk-deployment/scripts/v1.2/extract_version.py --branch_name $CIRCLE_BRANCH)
```
Comment (sravan-s) [inline .circleci/config.yml:73]: need small clarification, is this gonna be `release/v3.2.4` or just `release/3.2.4`?
good to comment here

File: .circleci/config.yml [hunk 3]
```diff
@@ -43,10 +61,68 @@ jobs:
...
+            source $BASH_ENV
+      - run:
+          name: Jira - Create release version
+          command: |
+            python3 ~/sdk-deployment/scripts/v1.2/create_version.py\
+              --jira_auth_user $JIRA_AUTH_USER\
+              --jira_auth_api_token $JIRA_AUTH_API_TOKEN\
+              --project SDKRLSD\
+              --name ios_core@$RELEASE_VERSION\
+              --description 'Created by automation'
```
Comment (sravan-s) [inline .circleci/config.yml:84]: Created by automation @ repo_name

File: .circleci/config.yml [hunk 4]
```diff
@@ -43,10 +61,68 @@ jobs:
...
+            echo "export RELEASE_VERSION=$release_version" >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: Jira - Create release version
+          command: |
+            python3 ~/sdk-deployment/scripts/v1.2/create_version.py\
+              --jira_auth_user $JIRA_AUTH_USER\
+              --jira_auth_api_token $JIRA_AUTH_API_TOKEN\
+              --project SDKRLSD\
+              --name ios_core@$RELEASE_VERSION\
```
Comment (sravan-s) [inline .circleci/config.yml:83]: name: js_uikit@$RELEASE_VERSION

File: .circleci/config.yml [hunk 5]
```diff
@@ -17,6 +27,14 @@ commands:
           name: UIKit for React test
           command: yarn run test
           no_output_timeout: 15m
+  clone_deployment_script:
+    steps:
+      - run:
+          name: Git - Clone deployment scripts
+          when: always
+          command: |
+            git config --global user.email "sha.sdk_deployment@sendbird.com"
```
Comment (sravan-s) [inline .circleci/config.yml:36]: need to rethink this email, this necessary?

File: .circleci/config.yml (hunk 4 above)
Comment (sravan-s) [reply .circleci/config.yml:83]: Similar, naming for actions below

File: .circleci/config.yml (hunk 5 above)
Comment (AhyoungRyu) [reply .circleci/config.yml:36]: You mean this line itself or just sdk_deployment email?

File: .circleci/config.yml (hunk 4 above)
Comment (AhyoungRyu) [reply .circleci/config.yml]: Oh I was a bit clumsy 😅 Replaced in here https://github.com/sendbird/sendbird-uikit-react/pull/658/commits/0913283ced05d7ab0763586bb678271b812908f2

File: .circleci/config.yml (hunk 3 above)
Comment (AhyoungRyu) [reply .circleci/config.yml]: Handled in this commit https://github.com/sendbird/sendbird-uikit-react/pull/658/commits/0913283ced05d7ab0763586bb678271b812908f2 too

File: .circleci/config.yml (hunk 2 above)
Comment (AhyoungRyu) [reply .circleci/config.yml:76]: I set a validation to make the version name start with `release/v`. 
The gh workflow will only be executed when the branch name starts wth `release/v`. 
But I probably need to add a guidance to let ppl know this in some say 
https://github.com/sendbird/sendbird-uikit-react/pull/658/commits/49255ab8b33b8144b9dffda32553bad820eb3c8d

File: .circleci/config.yml (hunk 1 above)
Comment (AhyoungRyu) [reply .circleci/config.yml]: Yeah made a change in here https://github.com/sendbird/sendbird-uikit-react/pull/658/commits/0a0ca993b99cdcbb1611002c151de28229d6daa1

File: .circleci/config.yml (hunk 5 above)
Comment (sravan-s) [reply .circleci/config.yml:41]: sha.sdk_deployment@sendbird.com, > curious why its there

File: .github/workflows/release-ticket-creation.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    inputs:
+      branch_name:
+        description: 'Release target branch name'
+        required: true
+
+jobs:
+  trigger-release-ticket-creation:
+    name: Trigger release ticket creation
+    runs-on: ubuntu-latest
+    # The branch name should starte with 'release/v'
```
Comment (sravan-s) [inline .github/workflows/release-ticket-creation.yml:13]: 👍

**Review by sravan-s — APPROVED** [review]
LGTM

**Review by HoonBaek — APPROVED** [review]

File: .github/workflows/release-ticket-creation.yml (hunk 6 above)
Comment (AhyoungRyu) [reply .github/workflows/release-ticket-creation.yml:13]: Oh my there's a typo 
```suggestion
    # The branch name should starts with 'release/v'
```

File: .circleci/config.yml (hunk 5 above)
Comment (AhyoungRyu) [reply .circleci/config.yml:41]: I guess the repo that we're trying to clone [here](https://github.com/sendbird/sdk-deployment) is private one so, need to setup a credential by setting the user.email like this.

Comment (AhyoungRyu) [reply .circleci/config.yml:41]: But I think, it's better to hide this info from here. Let me set it as a env variable.

Comment (AhyoungRyu) [reply .circleci/config.yml]: Noticed that unlike other chat SDK repositories, our is public. So I save the sdk user email as a secret env variable.
https://github.com/sendbird/sendbird-uikit-react/pull/658/commits/69d705a06c1b957159d56f58138978dcb396d6dc

**Comment by AhyoungRyu** [comment]
To test [this `workflow_dispatch` trigger method](https://github.com/sendbird/sendbird-uikit-react/pull/658/files#diff-108110a4e6452dafabe4ed6851dcf18a28d3662437484cff75e92c0107cbfcf8R3), [the file should be in the default base branch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch); `main` branch. 
So let me merge this into the main branch. I'll test this more from my side and will fix if it doesn't work as I expected.

**Pull request merged**